langchain/evaluation/run_evaluators/string_run_evaluator.py

```diff
@@ -383,3 +398,71 @@ def from_model_and_evaluator(
             example_mapper=example_mapper,
             string_evaluator=evaluator,
         )
+
+    @classmethod
+    def from_run_and_data_type(
+        cls,
+        evaluator: StringEvaluator,
+        run_type: RunTypeEnum,
+        data_type: DataType,
+        input_key: Optional[str] = None,
+        prediction_key: Optional[str] = None,
+        reference_key: Optional[str] = None,
+        tags: Optional[List[str]] = None,
+    ) -> StringRunEvaluatorChain:
+        """
+        Create a StringRunEvaluatorChain from an evaluator and the run and dataset types.
+
+        This method provides an easy way to instantiate a StringRunEvaluatorChain, by
+        taking an evaluator and information about the type of run and the data.
+        The method supports LLM and chain runs.
+
+        Args:
+            evaluator (StringEvaluator): The string evaluator to use.
+            run_type (RunTypeEnum): The type of run being evaluated.
+                Supported types are LLM and Chain.
+            data_type (DataType): The type of dataset used in the run.
+            input_key (str, optional): The key used to map the input from the run.
+            prediction_key (str, optional): The key used to map the prediction from the run.
+            reference_key (str, optional): The key used to map the reference from the dataset.
+            tags (List[str], optional): List of tags to attach to the evaluation chain.
+
+        Returns:
+            StringRunEvaluatorChain: The instantiated evaluation chain.
+
+        Raises:
+            ValueError: If the run type is not supported, or if the evaluator requires a
+                reference from the dataset but the reference key is not provided.
+
+        """  # noqa: E501
+
+        # Configure how run inputs/predictions are passed to the evaluator
+        if run_type == RunTypeEnum.llm:
+            run_mapper: StringRunMapper = LLMStringRunMapper()
+        elif run_type == RunTypeEnum.chain:
+            run_mapper = ChainStringRunMapper(
+                input_key=input_key, prediction_key=prediction_key
```

Comment:
what happens when the input and prediction keys are None?

Reply:
Currently the single key inference is happening in the runner utils rather than in the run mapper itself, so it would err. I'll update to handle the single-key case as well